### PR TITLE
Implement bulk renewal contract

### PR DESCRIPTION
**Pull request by Arachnid**
This contract makes it possible to renew multiple domains in a single TX.

### CODE REVIEW

File: contracts/BulkRenewal.sol [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    function renewAll(string[] calldata names, uint duration) external payable {
+        ETHRegistrarController controller = getController();
+        for(uint i = 0; i < names.length; i++) {
+            uint cost = controller.rentPrice(names[i], duration);
+            controller.renew.value(cost)(names[i], duration);
+        }
+        // Send any excess funds back
+        msg.sender.transfer(address(this).balance);
```
Comment (makoto) [inline contracts/BulkRenewal.sol]: Can we add a test case to assert this behavior?

**Comment by makoto** [comment]
@Arachnid tests seem failing

**Comment by decanus** [comment]
@Arachnid fix tests then I'll look.

**Comment by Arachnid** [comment]
@makoto @decanus PTAL

File: contracts/BulkRenewal.sol [hunk 2]
```diff
@@ -0,0 +1,48 @@
+pragma solidity >=0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "@ensdomains/ens/contracts/ENS.sol";
+import "./ETHRegistrarController.sol";
+import "@ensdomains/resolver/contracts/Resolver.sol";
+
+contract BulkRenewal {
+    bytes32 constant private ETH_NAMEHASH = 0x93cdeb708b7545dc668eb9280176169d1c33cfd8ed6f04690a0bcc88a93fc4ae;
```
Comment (decanus) [inline contracts/BulkRenewal.sol]: I usually like having the value of the namehash here like actually the `keccak` func call.

File: test/test_bulk_renewal.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+const namehash = require('eth-ens-namehash');
+const sha3 = require('web3-utils').sha3;
+const toBN = require('web3-utils').toBN;
+
+const NULL_ADDRESS = "0x0000000000000000000000000000000000000000"
+const ETH_LABEL = sha3('eth');
+const ETH_NAMEHASH = namehash.hash('eth');
+
+const advanceTime = Promise.promisify(function(delay, done) {
```
Comment (decanus) [inline test/test_bulk_renewal.js]: we have a library for this type of stuff. https://github.com/ensdomains/test-utils

File: test/test_bulk_renewal.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	web3.currentProvider.send({
+		jsonrpc: "2.0",
+		"method": "evm_increaseTime",
+		params: [delay]}, done)
+	}
+);
+
+async function expectFailure(call) {
```
Comment (decanus) [inline test/test_bulk_renewal.js]: ditto

File: contracts/BulkRenewal.sol (hunk 2 above)
Comment (Arachnid) [reply contracts/BulkRenewal.sol]: @decanus In this case it's a nested series of calls - so a little cumbersome to do.

Comment (decanus) [reply contracts/BulkRenewal.sol]: ok fair

**Pull request closed without merging**